scripts-dev/complement.sh

```diff
@@ -158,7 +158,10 @@ else
 
   # We only test faster room joins on monoliths, because they are purposefully
   # being developed without worker support to start with.
-  test_tags="$test_tags,faster_joins"
+  #
+  # Importing historical history and jump to date also only support monoliths,
+  # currently.
+  test_tags="$test_tags,faster_joins,msc2716,msc3030"
```

Comment:
@MadLittleMods Do you know if these are *expected* to work fine with workers or not?

Reply:
(I somewhat expect the jump-to-date one might just be a race in the complement tests, not sure though.)

Reply:
They've worked fine for a while now in CI 🤷‍♀️

I don't know what failure your talking about though. There is a flake https://github.com/matrix-org/synapse/issues/13944 that was seen in the MSC3030 tests

Reply:
> They've worked fine for a while now in CI 🤷‍♀️

Current `develop` doesn't send `/messages` (and a handful of other endpoints) to a worker in complement, so I think this might not actually have been working?

> I don't know what failure your talking about though. There is a flake https://github.com/matrix-org/synapse/issues/13944 that was seen in the [MSC3030](https://github.com/matrix-org/matrix-spec-proposals/pull/3030) tests

Sorry, should have linked to the failing run: https://github.com/matrix-org/synapse/actions/runs/3182998161/jobs/5189731097; looking at that a bit now though I wonder if it was due to timeouts though? I kicked it off again: https://github.com/matrix-org/synapse/actions/runs/3182998161/jobs/5191751580

Reply:
> Do you know if these are expected to work fine with workers or not?
>
> Current develop doesn't send /messages (and a handful of other endpoints) to a worker in complement, so I think this might not actually have been working?

Oh, I see. I would expect them to work in worker mode. So we might just need to fix up the test flakiness.

---

They look like legit failures.

For MSC3030, we probably need to make this retry and use `fetchUntilMessagesResponseHas`, https://github.com/matrix-org/complement/blob/0b7d990ca22b0ba9beeaf55534e3fb2413cfc654/tests/msc3030_test.go#L206-L220

For MSC2716 with the `JSONFieldStr: key 'end' missing` stuff, I probably need to separate out and merge the change from https://github.com/matrix-org/complement/pull/214/files#diff-9868c2480f7570d7028a300cdf808ea451dee2ba46adc5a0bfd5fb0b5951a2eaR1370-R1374. And I suspect there is another error underlying that where we maybe need to wait for replication to finish somehow.

I can take a look at this today 👍

Reply:
Re-running them provided the same failures. 😢

Reply:
@clokep This PR looks to be passing now ✅. Did this eventually flake back to passing?

Reply:
> I still want to look at those failures though.

Will sync with you today about this! I think going ahead and merging this still might make sense since it:

1. Tests a bunch of existing endpoints (parity with sytest).
2. Enables `/relations`.
3. Gives a way forward to test the failing feature.

I'm not in a huge rush to finish this though (the work has already been pending for several months...) Let me know what you're thinking!

Reply:
@clokep It seems like this PR can be merged and we can iterate on the MSC2716/MSC3030 tests separately, then re-enable the tags in a separate PR 👍